pkg/apps/watcher.py

```diff
@@ -0,0 +1,125 @@
+import os
+import sys
+import ctypes
+import struct
+
+# Our own little abstraction on top of inotify.  This only supports
+# watching directories non-recursively, but it also supports watching
+# directories that come and go into and out of existence.
```

Comment:
What about using [GFileMonitor](https://developer.gnome.org/gio/stable/GFileMonitor.html) through GI? cockpit-packagekit already transitively (through PackageKit, I suppose) pulls in `gobject-introspection` which pulls in the Gio typelib, and with that PR we'll already need python or python3, so the only missing dependency for that would be `python3-gobject-base` (298 kB, no extra dependencies - tested in a minimal mock chroot).

Reply:
Can `GFileMonitor` monitor non-existing directories?  That was the whole point of this excersize.

Also, I think we need to monitoring functionality without PackageKit, when we handle containers on the Atomics.

Reply:
No, I don't think it can. The usual approach is to pre-create the directory and then watch it. Good point for Atomic, we can't rely on GI there (but can we rely on having Python?). So keep this for now.

Reply:
I have to admit I wanted to see what it takes to support watching files/directories that come and go.  File watching itself is dubious, but still ... I am a bit annoyed that there isn't any decent code out there for it.